packages/console/src/pages/UserDetails/UserSettings/components/UserSsoIdentities/index.tsx

```diff
@@ -0,0 +1,112 @@
+import type { SsoConnectorWithProviderConfig, UserSsoIdentity } from '@logto/schemas';
+import { useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import useSWR from 'swr';
+
+import CopyToClipboard from '@/ds-components/CopyToClipboard';
+import ImageWithErrorFallback from '@/ds-components/ImageWithErrorFallback';
+import Table from '@/ds-components/Table';
+import type { RequestError } from '@/hooks/use-api';
+
+import * as styles from '../UserSocialIdentities/index.module.scss';
+
+type Props = {
+  ssoIdentities: UserSsoIdentity[];
+};
+
+type DisplayConnector = {
+  logo: SsoConnectorWithProviderConfig['providerLogo'];
+  name: SsoConnectorWithProviderConfig['connectorName'];
+  userIdentity: UserSsoIdentity['identityId'];
+  issuer: UserSsoIdentity['issuer'];
+};
+
+function UserSsoIdentities({ ssoIdentities }: Props) {
+  const { t } = useTranslation(undefined, {
+    keyPrefix: 'admin_console',
+  });
+
+  const { data, error, mutate } = useSWR<SsoConnectorWithProviderConfig[], RequestError>(
+    'api/sso-connectors'
+  );
+
+  const isLoading = !data && !error;
+
+  const displaySsoConnectors = useMemo(() => {
+    if (!data) {
+      return;
+    }
+
+    // eslint-disable-next-line no-restricted-syntax
+    return ssoIdentities
+      .map((identity) => {
+        const { providerLogo: logo, connectorName: name } =
+          data.find((ssoConnector) => ssoConnector.id === identity.ssoConnectorId) ?? {};
+        const { identityId: userIdentity, issuer } = identity;
+
+        if (!(logo && name)) {
+          return;
+        }
+
+        return { logo, name, userIdentity, issuer };
+      })
+      .filter(Boolean) as DisplayConnector[];
```

Comment:
nit: `.filter((identity): identity is DisplayConnector => Boolean(identity))`

Reply:
Updated